---
title: "Human integration"
author: "James Ashmore"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this analysis we are going to perform a cross-species integration.

## Setup

Define chunk options:

```{r}
knitr::opts_chunk$set(
  autodep    = TRUE,
  cache      = TRUE,
  cache.path = "cache/11-human-integration.Rmd/",
  dev        = "png",
  error      = FALSE,
  message    = FALSE,
  warning    = FALSE,
  fig.align = "center",
  out.width = '100%'
)
```

Attach required packages:

```{r}
library(batchelor)
library(ggforce)
library(ggrepel)
library(scater)
library(scran)
library(scuttle)
```

Import mouse and human experiment data:

```{r}
sce <- list(
  mouse = readRDS("output/09-cell-annotation.rds"),
  human = readRDS("data/HumanGastrulationData.rds")
)
```

Import one-to-one homologous gene annotations:

```{r}
ann <- read.csv("data/hsapiens_mmusculus.csv")
```

## Processing

Make cell names unique:

```{r}
colnames(sce$mouse) <- paste0("mouse-", seq_len(ncol(sce$mouse)))

colnames(sce$human) <- paste0("human-", seq_len(ncol(sce$human)))
```

Rename mouse cluster labels:

```{r}
sce$mouse$cluster <- paste("Cluster", sce$mouse$cluster)
```

Create mouse and human colour palettes:

```{r}
MouseClusterColours <- c(
  "Cluster 1" = "#6388B4",
  "Cluster 2" = "#55AD89",
  "Cluster 3" = "#BB7693",
  "Cluster 4" = "#8CC2CA"
)

MouseCelltypeColours <- MouseGastrulationData::EmbryoCelltypeColours

HumanClusterColours <- c(
  "Epiblast" = "#3D7AB0",
  "Ectoderm" = "#4FC0CB",
  "Primitive Streak" = "#A6BEDF",
  "Nascent Mesoderm" = "#C67DB4",
  "Axial Mesoderm" = "#7B534D",
  "Emergent Mesoderm" = "#ABB973",
  "Endoderm" = "#F28941",
  "Advanced Mesoderm" = "#4F9E74",
  "YS Mesoderm" = "#FAB781",
  "Hemogenic Endothelial Progenitors" = "#6A59A6",
  "Erythrocytes" = "#C5433E"
)

HumanOriginColours <- c(
  "Caudal" = "#4DAF4A",
  "Rostral" = "#377EB8",
  "Yolk Sac" = "#FF7F00"
)

BatchColours <- c(
  "mouse" = "#00A2B3",
  "human" = "#F7C480"
)
```

Create human label positions:

```{r}
HumanClusterLabels <- makePerCellDF(sce$human, use.coldata = FALSE, use.dimred = "UMAP")

HumanClusterLabels <- aggregate(HumanClusterLabels, list(Cluster = sce$human$Cluster), mean)
```

Create batch facet labels:

```{r}
BatchLabels <- c("human" = "Gastrulating embryo (Human)", "mouse" = "Neuromesodermal Progenitors (Mouse)")
```

## Exploration

Plot mouse cluster annotation:

```{r}
ggcells(sce$mouse, aes(UMAP.1, UMAP.2, colour = cluster)) + 
  geom_point() + 
  scale_colour_manual(values = MouseClusterColours) + 
  ggtitle("Mouse Neuromesodermal Progenitors") + 
  theme_no_axes(theme_bw()) + 
  theme(aspect.ratio = 1, 
        legend.title = element_blank(),
        legend.position = c(0.01, 0.99),
        legend.justification = c(0, 1))
```

Plot mouse celltype annotation:

```{r}
ggcells(sce$mouse, aes(UMAP.1, UMAP.2, colour = celltype)) + 
  geom_point() + 
  scale_colour_manual(values = MouseCelltypeColours) + 
  ggtitle("Mouse Neuromesodermal Progenitors") + 
  theme_no_axes(theme_bw()) + 
  theme(aspect.ratio = 1, 
        legend.title = element_blank(),
        legend.position = c(0.01, 0.99),
        legend.justification = c(0, 1))
```

Plot human embryo cluster annotation:

```{r}
ggcells(sce$human, aes(UMAP.1, UMAP.2, colour = Cluster)) + 
  geom_point(size = 0.5) + 
  geom_text_repel(
    data = HumanClusterLabels, 
    mapping = aes(label = Cluster), 
    size = 3, 
    colour = "#000000") + 
  scale_colour_manual(values = HumanClusterColours) + 
  theme_no_axes(theme_bw()) + 
  theme(aspect.ratio = 1,
        legend.position = "none")
```

Plot human embryo origin annotation:

```{r}
ggcells(sce$human, aes(UMAP.1, UMAP.2, colour = Origin)) + 
  geom_point(size = 0.5) + 
  scale_colour_manual(values = HumanOriginColours) + 
  theme_no_axes(theme_bw()) + 
  theme(aspect.ratio = 1,
        legend.title = element_blank(),
        legend.position = c(0.99, 0.01),
        legend.justification = c(1, 0))
```

## Integration

Filter experiment data by gene symbol:

```{r}
sce$mouse <- sce$mouse[rowData(sce$mouse)$gene_name %in% ann$Mouse_Symbol, ]

sce$human <- sce$human[rowData(sce$human)$Symbol %in% ann$Human_Symbol, ]
```

Rename mouse data by human gene symbol:

```{r}
rownames(sce$mouse) <- ann$Human_Symbol[match(rowData(sce$mouse)$gene_name, ann$Mouse_Symbol)]
```

Filter experiment data to common gene symbol:

```{r}
use <- intersect(rownames(sce$mouse), rownames(sce$human))

sce$mouse <- sce$mouse[use, ]

sce$human <- sce$human[use, ]
```

Calculate variance statistics:

```{r}
dec <- list(
  mouse = modelGeneVar(sce$mouse),
  human = modelGeneVar(sce$human)
)
```

Combine variances to obtain a single set of highly variable genes:

```{r}
dec <- combineVar(dec$mouse, dec$human)

hvg <- getTopHVGs(dec, n = 2000)
```

Adjust for gross differences in sequencing depth:

```{r}
sce <- multiBatchNorm(
  mouse = sce$mouse,
  human = sce$human
)
```

Perform nearest neighbors correction:

```{r}
set.seed(880353265)

mnn <- fastMNN(sce, subset.row = hvg, merge.order = list("human", "mouse"), correct.all = TRUE)
```

Combine column data from both experiments:

```{r}
mnn$Batch <- mnn$batch; mnn$batch <- NULL

mnn$Cluster <- c(sce$mouse$cluster, sce$human$Cluster)

mnn$Origin <- c(rep(NA, times = ncol(sce$mouse)), sce$human$Origin)
```

## Diagnostics

Calculate UMAP matrix:

```{r}
set.seed(161773252)

mnn <- runUMAP(mnn, n_neighbors = 20, min_dist = 0.5, dimred = "corrected")
```

Plot merged batch annotation:

```{r}
ggcells(mnn, aes(UMAP.1, UMAP.2, colour = Batch)) + 
  geom_point(size = 0.5) + 
  scale_colour_manual(values = BatchColours, na.value = "#000000") + 
  theme_no_axes(theme_bw()) + 
  theme(aspect.ratio = 1,
        legend.title = element_blank(),
        legend.position = c(0.99, 0.01),
        legend.justification = c(1, 0))
```

Plot merged origin annotation:

```{r}
ggcells(mnn, aes(UMAP.1, UMAP.2, colour = Origin)) + 
  geom_point(size = 0.5) + 
  scale_colour_manual(values = HumanOriginColours, na.value = "#000000") + 
  theme_no_axes(theme_bw()) + 
  theme(aspect.ratio = 1,
        legend.title = element_blank(),
        legend.position = c(0.99, 0.01),
        legend.justification = c(1, 0))
```

Plot merged cluster annotation:

```{r fig.width = 12, fig.height = 9}
ggcells(mnn, aes(UMAP.1, UMAP.2, colour = Cluster)) + 
  geom_point(data = makePerCellDF(mnn, use.coldata = FALSE, use.dimred = "UMAP"), colour = "gainsboro", size = 1) + 
  geom_point(size = 1) + 
  scale_colour_manual(values = c(HumanClusterColours, MouseClusterColours), na.value = "#000000") + 
  facet_wrap(~ Batch, labeller = labeller(Batch = BatchLabels)) + 
  theme_no_axes(theme_bw()) + 
  theme(aspect.ratio = 1,
        legend.title = element_blank(),
        strip.background = element_blank())
```

## Summary

### Output

Save merged experiment object to disk:

```{r}
saveRDS(mnn, file = "output/11-human-integration.rds")
```

### Session

Print session information:

```{r}
sessionInfo()
```
